### PR TITLE
Inject additional dependencies into the serializers

### PR DESCRIPTION
**Pull request by mlw**
The serializers were using static variables in a couple of places that was fine for production, but made some types of tests difficult since the static variables would outlive the mocks.

This change should have no functional impact on Santa - nothing should change at runtime as part of this restructuring to inject the dependencies.

### CODE REVIEW

**Comment by mlw** [comment]
Please hold up on review - going to store the client mode in the SNTCachedDecision so that lazy client mode lookup isn't needed (also fixes a long standing race that could occur where the mode applied at policy time might not match the mode at logging time, tho this would be quite rare).

**Comment by mlw** [comment]
PR updated to store the client mode at decision time and remove the client mode lookup function injection.

Good to resume reviewing!

File: Source/common/SNTCachedDecision.h [hunk 1]
```diff
@@ -42,4 +42,6 @@
 @property NSString *customMsg;
 @property BOOL silentBlock;
 
+@property SNTClientMode decisionClientMode;
```
Comment (russellhancox) [inline Source/common/SNTCachedDecision.h:45]: Group this with the decision further up

**Review by russellhancox — CHANGES_REQUESTED** [review]
LGTM

**Review by russellhancox — APPROVED** [review]

**Pull request merged**